src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,362 @@
+			CSVPrinter identifiersWithNoReferrerPrinter = new CSVPrinter(new FileWriter(COSMICUpdateUtil.reportsDirectoryPath  + File.separator + "COSMICIdentifiersNoReferrers_"+dateSuffix+".csv"), CSVFormat.DEFAULT.withHeader("COSMIC identifier")))
+		{
+			for (GKInstance cosmicObject : cosmicObjects)
+			{
+				String identifier = (String)cosmicObject.getAttributeValue(ReactomeJavaConstants.identifier);
+				COSMICIdentifierUpdater updater = new COSMICIdentifierUpdater();
+				updater.setIdentifier(identifier);
+				updater.setDbID(cosmicObject.getDBID());
+
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> EWASes = cosmicObject.getReferers(ReactomeJavaConstants.crossReference);
+				// If NO EWASes exist, then log this information.
+				if (EWASes == null || EWASes.isEmpty())
+				{
+					identifiersWithNoReferrerPrinter.printRecord(identifier);
+				}
+				else
+				{
+					// Check the EWASes for mismatches between referenceSequence identifier and main identifier.
+					// If there's a mismatch then suggest COSF, else suggest COSM.
+					checkEWASes(nonEWASPrinter, identifier, updater, EWASes);
+				}
+				
+				// support function for populating list.
+				BiFunction<? super String, ? super List<COSMICIdentifierUpdater>, ? extends List<COSMICIdentifierUpdater>> listPopulator = (k,v) -> {
+					if (v == null)
+					{
+						List<COSMICIdentifierUpdater> updaters = new ArrayList<>();
+						updaters.add(updater);
+						return updaters;
+					}
+					else
+					{
+						v.add(updater);
+						return v;
+					}
+				};
+				
+				// If the identifier starts with C it's not a numeric identifier.
+				if (COSMICUpdateUtil.stringStartsWithC(identifier))
+				{
+					updates.compute(updater.getIdentifier(), listPopulator );
+					
+				}
+				else
+				{
+
+					updates.compute(updater.getSuggestedPrefix() + updater.getIdentifier(), listPopulator );
+				}
+			}
+		}
+		return updates;
+	}
+
+	/**
+	 * Checks EWASes to see if they have modifiedResidues that have a referenceSequence that is NOT the same as the EWASes referenceEntity.
+	 * The suggested prefix will be set to COSF on the update record if mismatches are found, otherwise COSM will be set.
+	 * @param nonEWASPrinter CSVPrinter for reporting.
+	 * @param identifier Identifier of the object being checked, used for reporting.
+	 * @param updater A COSMICIdentifierUpdater whose suggested prefix will be updated.
+	 * @param EWASes The EWASes to check. If a non-EWAS is in this list, it will be reported.
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 * @throws IOException
+	 */
+	private static void checkEWASes(CSVPrinter nonEWASPrinter, String identifier, COSMICIdentifierUpdater updater, Collection<GKInstance> EWASes) throws InvalidAttributeException, Exception, IOException
+	{
+		String prefix;
+		GKInstance[] EWASArray = EWASes.toArray(new GKInstance[0]);
+		boolean done = false;
+		int i = 0;
+		while(!done)
+		{
```

Comment:
You could get rid of the `done`, `i`, and `EWASArray` variables by using getting an iterator on the `EWASes` Collection and doing: 

`while(EWASesIterator.hasNext() && !foundMismatchedRefSequence)`

It might be easier and clearer than navigating the indexes explicitly and you can have the end condition in the while loop itself.

Reply:
I find the explicitness of this easier to read/comprehend.

Reply:
Okay